packages/typegen/src/types.ts

```diff
@@ -44,12 +44,23 @@ export interface Options {
 
   /**
    * Files to look for SQL queries in. e.g. `source/queries/*.ts`
-   * Also allows passing `cwd` and `ignore` strings e.g. `['source/*.ts', {ignore: ['source/*.test.ts']}]`
-   * Defaults to all '.ts' and '.sql' files, ignoring node_modules.
-   *
-   * Use the `{since: string}` format to match files changed since the given git ref. e.g. `{since: 'main'}` or `{since: 'HEAD~1'}`
+   * Defaults to all '.ts' and '.sql' files.
+   */
+  glob: string
+
+  /**
+   * Glob filename pattern or array of patterns to ignore e.g. `source/don-touch/*.ts`
+   * Defaults to ignore node_modules.
+   */
+  ignore: string | string[]
+
+  /**
+   * Filter matcher results to files from specific git refs.
+   * Use to match files changed since the given git ref. e.g. `'main'` or `'HEAD~1'`
+   * This option has no effect in watch mode, as every file change constitutes a git diff.
+   * Defaults to undefined (git filter disabled)
    */
-  glob: string | [string, {ignore?: string[]}?] | {since: string}
```

Comment:
I do like this change - but the reason for the `[string, { ignore?: string[] }?]` was because it matches `Parameters<typeof glob>`. I was vaguely thinking about how to be forwards-compatible with [glob options](https://www.npmjs.com/package/glob#properties) which has various advanced options to do with caching/symlinking etc. I haven't had a use-case personally yet, but didn't want to paint the library into a corner. In fact, since we were using `{ ...globParams[1] }` it's not impossible that some users are already passing unsupported options. I'm not philosophically against breaking people doing that since it's an undocumented hack, but it is kind of a legit use-case, I think.

Going back to the original two problems this is solving:

1. Ignore was not configurable via CLI
2. Both the glob pattern as well as the ignore pattern were discarded, when using since.

Off the top of my head it seems to me both could be solved by a smaller breaking change - just moving `since` out to a separate property and keeping `glob` defined as `string | [string, {ignore?: string[]}?]`. We could still add the `--ignore` CLI parameter. Could that work maybe?

Reply:
Yes I figured your intention from the code.

However even in glob they are individual parameters, not a tuple.
My point is that `glob` and `options` should be separate, as they are in `glob`...

I would doubt that anyone is currently using other glob options, because it is absolutely not obvious externally that this is possible, because A) As described above the tuple here is not a standard interface and the option isn't called `globOptions` or smth. B) the exposed types even forbid it, so anyone using TS would have to use as `any` to even assign it to the glob option.
So the only ones attempting this would probably be you or me.

If you feel strongly about this, we could expose a [glob options compatible object](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/glob/index.d.ts#L29) instead of `ignore` in the typegen config for example `globOptions`.
Its type would be an object with currently only one prop: `ignore`
Then we could overwrite that property's value with the CLI params from ignore.

I would still vote against this though, because:
1. I like that CLI and config are in sync for the most part. It makes it easier to understand what is equivalent to what.
2. What about since? Technically you'd also consider it an option affecting what get's matched. So shouldn't it also be part of `globOptions`? But now we're breaking compatibility with glob...
3. We would still have to pick specific properties, because not all options make sense and some options are set by the script i.e. cwd, absolute... Otherwise it could cause unexpected behaviour or obfuscate why some options make it through, while others don't.
4. I like configs to be as flat as possible (subjective).
5. It's still no less breaking.

Have you thought about renaming `glob` to `match` or `pattern`?
This would make this breaking change even more obvious.

Reply:
Yeah, this is all very fair - I guess we need to decide if this is more a callable js API, or a CLI. I think of it as a library with an API, because it's biased towards configuring in js then just running `slonik-typegen`. But it's not unreasonable to try to make everything flat which is intuitive in both config modes.

I would prefer to rename `glob` to make the breaking change more obvious, yes. Ok so to continue to bikeshed on what the properties should be, some thoughts. Maybe `include` and `exclude` to match tsconfig.json? Later if we needed to we could allow for `include` to be a function returning `Promise<string[]>` so that if you really wanted you could call `await glob(anythingYouLike, ...)`. But we could not bother with that until someone actually asks for it.

One further thought - with this change, do we still need `rootDir`? It was only really important for limiting `since` to relevant files, but now that you've improved it so since only applies to the files matched by `glob` (or whatever it gets renamed to) maybe we could drop it?

Also if we change this, to avoid noise maybe worth adding some validation to the config since we're on v0. Could be some simple `assert.ok(!('glob' in options), `The 'glob' option is deprecated. Please use 'include' instead`). Orrrr I guess we could go to v1. But with the big-ish changes you've brought up in #385 and #386 I think we shouldn't.

Reply:
`include` and `exclude` - yes no longer piggy backing off glob but tsconfig will be more familiar to most most users, so let’s go for it if you are ok with it. 

`rootDir` - let’s leave it for this PR. `rootDir` also matches tsconfig, not `baseDir`, you might be mixed up with `baseUrl`. Either way let’s not touch, I think there are ways to achieve what you mention without it but not related to this PR.